lib/ansible/plugins/inventory/tower.py

```diff
@@ -0,0 +1,203 @@
+# The inventory path must always be @tower_inventory if you are reading all settings from environment variables.
+# ansible-inventory -i @tower_inventory --list
+'''
+
+import re
+import os
+import json
+from ansible.module_utils import six
+from ansible.module_utils.urls import Request, urllib_error, ConnectionError, socket, httplib
+from ansible.module_utils._text import to_text, to_native
+from ansible.errors import AnsibleParserError
+from ansible.plugins.inventory import BaseInventoryPlugin
+
+# Python 2/3 Compatibility
+try:
+    from urlparse import urljoin
+except ImportError:
+    from urllib.parse import urljoin
+
+
+class InventoryModule(BaseInventoryPlugin):
+    NAME = 'tower'
+    # Stays backward compatible with tower inventory script.
+    # If the user supplies '@tower_inventory' as path, the plugin will read from environment variables.
+    read_all_configs_from_env_var = False
+
+    def read_tower_inventory(self, tower_host, tower_user, tower_pass, inventory, license_type, verify_ssl=True):
+        if not re.match('(?:http|https)://', tower_host):
+            tower_host = "https://{tower_host}".format(tower_host=tower_host)
+        inventory_id = inventory.replace('/', '')
+        inventory_url = "/api/v2/inventories/{inv_id}/script/?hostvars=1&towervars=1&all=1".format(inv_id=inventory_id)
+        inventory_url = urljoin(tower_host, inventory_url)
+        config_url = urljoin(tower_host, "/api/v2/config/")
+
+        request_handler = Request(url_username=tower_user,
+                                  url_password=tower_pass,
+                                  force_basic_auth=True,
+                                  validate_certs=verify_ssl)
+
+        try:
+            if license_type != "open":
+                config_response = request_handler.get(config_url)
+                if config_response.getcode() == 200:
+                    config_json = json.loads(config_response.read())
+                    source_type = config_json['license_info']['license_type']
+                    if not source_type == license_type:
+                        error_string = to_native('Tower server licenses must match: source: {source} local: {local}'.
+                                                 format(source=source_type, local=license_type))
+                        raise AnsibleParserError(error_string)
+                else:
+                    raise AnsibleParserError(to_native('Failed to validate the license of the remote Tower: {response}'.
+                                             format(response=config_response)))
+
+            response = request_handler.get(inventory_url)
+            if response.getcode() != 200:
+                # If the GET /api/v2/inventories/N/script is not HTTP 200, print the error code
+                raise AnsibleParserError(to_native('Connection to remote host failed: {response}'
+                                                   .format(response=response)))
```

Comment:
Hey Alan I think if tower returns a non-200 status code, urllib will throw an exception which will be caught by the try-catch block. 

I would rather leave the message out as Tower sometimes return HTML as error message instead of a more "readable" format such as JSON or plain text.